client/pages/detail.jsx

```diff
@@ -46,7 +54,7 @@ export default class Detail extends React.Component {
                     </button>
                     <ul className="dropdown-menu dropdown-menu-end">
                       <li><a className="dropdown-item" href={`#edit?entryId=${this.props.entryId}`}>Edit</a></li>
-                      <li><a className="dropdown-item" href="#">Remove</a></li>
+                      <li><a className="dropdown-item" href="#" onClick={this.onClick}>Remove</a></li>
```

Comment:
Instead of making this an anchor, it should be a button (styled like an anchor). And in your click handler, assign the new hash to `window.location`

```js
fetch(...)
  .then(() => { window.location.hash = '#'; })
  .catch(err => console.error(err))
```